sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt

```diff
@@ -33,83 +33,37 @@ internal object KlaviyoApiClient : ApiClient {
      */
     private var apiObservers = mutableListOf<ApiObserver>()
 
-    init {
-        onApiRequest { r ->
-            when (r.state) {
-                Status.Unsent.name -> Registry.log.verbose("${r.type} Request enqueued")
-                Status.Inflight.name -> Registry.log.verbose("${r.type} Request inflight")
-                Status.PendingRetry.name -> {
-                    val attemptsRemaining = Registry.config.networkMaxAttempts - r.attempts
-                    Registry.log.warning(
-                        "${r.type} Request failed with code ${r.responseCode}, and will be retried up to $attemptsRemaining more times."
-                    )
-                }
-                Status.Complete.name -> Registry.log.verbose(
-                    "${r.type} Request succeeded with code ${r.responseCode}"
-                )
-                else -> Registry.log.error(
-                    "${r.type} Request failed with code ${r.responseCode}, and will be dropped"
-                )
-            }
-
-            r.responseBody?.let { response ->
-                val body = r.requestBody?.let { JSONObject(it).toString(2) }
-                Registry.log.verbose("${r.httpMethod}: ${r.url}")
-                Registry.log.verbose("Headers: ${r.headers}")
-                Registry.log.verbose("Query: ${r.query}")
-                Registry.log.verbose("Body: $body")
-                Registry.log.verbose("${r.responseCode} $response")
-            }
-        }
+    /**
+     * Initialize logic including lifecycle observers and reviving the queue from persistent store
+     */
+    override fun startService() {
+        Registry.lifecycleMonitor.offActivityEvent(::onLifecycleActivity)
+        Registry.lifecycleMonitor.onActivityEvent(::onLifecycleActivity)
 
-        // Stop our handler thread when all activities stop
-        Registry.lifecycleMonitor.onActivityEvent {
-            when (it) {
-                is ActivityEvent.AllStopped -> startBatch(true)
-                else -> Unit
-            }
-        }
+        Registry.networkMonitor.offNetworkChange(::onNetworkChange)
+        Registry.networkMonitor.onNetworkChange(::onNetworkChange)
 
-        // Stop the background batching job while offline
-        Registry.networkMonitor.onNetworkChange { isOnline ->
-            if (isOnline) {
-                startBatch(true)
-            } else {
-                stopBatch()
-            }
+        if (getQueueSize() == 0) {
+            // We only need to restore queue from persistent store once
+            restoreQueue()
```

Comment:
there isnt a case where we would have something in the queue but still want to restore from persistence? ie, conditionalizing on whether we've restored from persistence already isn't better/more logically correct?

Reply:
We only want to restore queue once, because after that we always keep the in-memory and persistent queue in sync. Previously we could be sure we only restored queue once, and it always happened first, because it was done from `init`. 
I think this is what you're saying too, but I'm thinking now that it'd be better/clearer here I used a single bool flag or lateinit var to make sure that it only happens once.